PQAnalysis/traj/trajectory.py

```diff
@@ -143,6 +148,93 @@ def __iter__(self) -> Iterable[AtomicSystem]:
             frame.topology = self.topology
             yield frame
 
+    def window(
+        self,
+        window_size: int,
+        window_gap: int = 1,
+        window_start: int = 0,
+        window_stop: int | None = None,
+    ) -> Iterable[List[AtomicSystem]]:
+        """
+        This method allows a window of the trajectory to be retrieved.
+        Window is a sequence of frames from start to stop with a step size and a gap size.
+
+        Parameters
+        ----------
+        window_size : int
+            The size of the window.
+        window_gap : int, optional
+            The gap size between two windows, by default 1
+        window_start : int, optional
+            The start index of the window, by default 0
+        window_stop : int, optional
+            The stop index of the window, by default None, which then
+            set to the length of the trajectory.
```

Comment:
still unclear weather stop is the end of the first window, or the start index of the last window or the last index of the last window